vega_sim/service.py

```diff
@@ -3645,18 +3645,30 @@ def update_margin_mode(
             margin_factor=str(margin_factor),
         )
 
-    def check_market_states_consistent(self, raise_exceptions: bool = True):
+    def check_market_states_consistent(
+        self, raise_exceptions: bool = True, max_attempts: int = 5
+    ):
+        exception = None
         self.wait_for_total_catchup()
-        markets = self.all_markets()
-        for market in markets:
-            market_data = self.get_latest_market_data(market_id=market.id)
-            try:
-                assert market.state == market_data.market_state
-                assert market.trading_mode == market_data.market_trading_mode
-            except AssertionError as e:
-                if raise_exceptions:
-                    raise e
-                logging.error(e)
+        for _ in range(max_attempts):
+            markets = self.all_markets()
+            for market in markets:
+                market_data = self.get_latest_market_data(market_id=market.id)
+                try:
+                    assert market.state == market_data.market_state
+                    assert market.trading_mode == market_data.market_trading_mode
+                    return
+                except AssertionError as e:
+                    exception = e
+                    time.sleep(5)
+                    logging.warning(
+                        "Check failed, waiting to ensure sync then retrying check."
+                    )
+                    self.wait_for_total_catchup()
+                    continue
+        if raise_exceptions:
+            raise exception
+        logging.error(exception)
```

Comment:
Best to have logging done through a `logger` defined with `get_logger` at the top of the module